cli.js

```diff
@@ -1,9 +1,29 @@
 #!/usr/bin/env node
 'use strict';
 const fs = require('fs');
+const path = require('path');
 const {getNativeImagePath, getFirstSupportedPlatform} = require('./lib/utils');
 const parseArgs = require('minimist');
 
+function mkDirByPathSync(targetDir, {isRelativeToScript = false} = {}) {
+  const sep = path.sep;
+  const initDir = path.isAbsolute(targetDir) ? sep : '';
+  const baseDir = isRelativeToScript ? __dirname : '.';
+
+  targetDir.split(sep).reduce((parentDir, childDir) => {
+    const curDir = path.resolve(baseDir, parentDir, childDir);
```

Comment:
I ran your test case and logged the output rather than create directories. Here's what was logged for the iterations:

```
/Users/chad/git/closure-compiler-npm/foo
/Users/chad/git/closure-compiler-npm/foo/bar
/Users/chad/git/closure-compiler-npm/foo/bar/baz
```

Which looks correct to me.

Reply:
OK, I must be misreading it then, or perhaps misunderstanding what path.resolve() does.